File: Model/Product.cs
namespace sep3.Model;

public class Product
{
    
    public double Price { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public string ImagePath { get; set; }
    public string Category { get; set; }
    public int Id { get; set; }
    public Brand Brand { get; set; }

    
    public Product()
    {
    }
    
    public Product(double price, string name, string description, string imagePath, string category, int id, Brand brand, string color, string size, string material, double discount)
    {
        this.Price = price;
        this.Name = name;
        this.Description = description;
        this.ImagePath = imagePath;
        this.Category = category;
        this.Id = id;
        this.Brand = brand;
    }
}

File: warehouse/src/main/java/sep3/warehouse/entities/Product.java
package sep3.warehouse.entities;


import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;


@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "product")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 255)
    private String name;

    @Column(nullable = false, length = 255)
    private String description;

    @Column(nullable = false)
    private double price;

    @Column(nullable = false)
    private String imagePath;

    @ManyToOne
    @JoinColumn(name = "brand_id")
    @JsonIgnoreProperties("products")
    private Brand brand;

    @JsonIgnoreProperties
    @OneToMany(mappedBy = "product", fetch = FetchType.LAZY)
    private List<ProductVariant> productVariants;
}
